components/pixel/schema.json

```diff
@@ -0,0 +1,42 @@
+{
+  "title": "Pixel Component Definition",
+  "description": "Addressable Pixel WipperSnapper component for use in Adafruit IO",
+  "type": "object",
+  "required": [ "displayName", "pixelsType", "defaultPixelsOrder"],
+  "properties": {
+    "displayName": {
+      "description": "The human-friendly name of this component.",
+      "type": "string",
+      "minLength": 3,
+      "maxLength": 24
+    },
+    "pixelsType": {
+        "description": "The type of addressable pixel: NEOPIXEL or DOTSTAR. Corresponds to `PixelsType` in pixels.proto",
+        "type": "string",
+        "pattern": "^(NEOPIXEL|DOTSTAR)$"
+    },
+    "defaultPixelsOrder": {
+        "description": "The pixel strand's color ordering. Corresponds to `defaultPixelsOrder` in pixels.proto.",
+        "type": "string",
+        "pattern": "^(GRB|GRBW|RGB|RGBW|BRG)$"
+    },
+    "defaultNumPixels": {
+        "description": "If present, the component form will allow the user to set the number of pixels in the strand, with this value as the default.",
```

Comment:
Unsure how this ends up working for my code.
- If _not_ present, we _don't_ let them set the number of pixels?
- Do we internally set one?
- Is the number used for anything yet?

Maybe you've already answered all this in basecamp or elsewhere!

Reply:
> If not present, we don't let them set the number of pixels?

This is a required field.

> Do we internally set one?

No

> Is the number used for anything yet?

Used by https://github.com/adafruit/Wippersnapper_Protobuf/pull/111/files#diff-a1604995e7a859acfac92b0c1873d95f017ae9712b241d7b1713035ab5ce6636R38